src/lib/concurrency/logging/abstract_recovery.cpp

```diff
@@ -0,0 +1,49 @@
+#include "abstract_recovery.hpp"
+
+#include "../../storage/storage_manager.hpp"
+#include "../../storage/table.hpp"
+#include "../transaction_manager.hpp"
+#include "types.hpp"
+#include "utils/load_table.hpp"
+
+namespace opossum {
+
+void AbstractRecovery::_redo_transactions(const TransactionID& transaction_id, std::vector<LoggedItem>& transactions) {
+  for (auto& transaction : transactions) {
+    if (transaction.transaction_id != transaction_id) continue;
+
+    auto table = StorageManager::get().get_table(transaction.table_name);
+    auto chunk = table->get_chunk(transaction.row_id.chunk_id);
+
+    if (transaction.type == LogType::Value) {
+      chunk->append(*transaction.values);
+
+      auto mvcc_columns = chunk->mvcc_columns();
+      DebugAssert(mvcc_columns->begin_cids.size() - 1 == transaction.row_id.chunk_offset,
+                  "recovery rowID " + std::to_string(mvcc_columns->begin_cids.size() - 1) + " != logged rowID " +
+                      std::to_string(transaction.row_id.chunk_offset));
+      mvcc_columns->begin_cids[mvcc_columns->begin_cids.size() - 1] = transaction_id;
```

Comment:
Commit IDs and Transaction IDs come from different domains. If you use a transaction id as a commit id, this needs to be documented.